pilot/cmd/pilot-agent/config.go

```diff
@@ -33,22 +33,24 @@ import (
 	"istio.io/pkg/log"
 )
 
-func constructProxyConfig() (meshconfig.ProxyConfig, error) {
+func constructProxyConfigAndMeshConfig() (meshconfig.ProxyConfig, meshconfig.MeshConfig, error) {
```

Comment:
I am not sure how this change can work - sidecars do not have access to mesh configuration.

Reply:
According the config.go, 
To get ProxyConfig, constructProxyConfig() also construct MeshConfig before. I just return created MeshConfig 
![image](https://user-images.githubusercontent.com/49465776/93038228-9bdcdd80-f66e-11ea-82fe-1c0d32ad0e27.png)

Reply:
It is just the default though. The sidecar only gets access to proxy config. Any change you make to read mesh config is just going to get the defaults 100% of the time